src/INode.java

```diff
@@ -229,6 +244,17 @@ private ASTValue evalDefineProperty(WordsEnvironment environment) {
 		throw new AssertionError("Not yet implemented");	
 	}
 
+	private ASTValue evalDivide(WordsEnvironment environment) throws WordsRuntimeException {
+		ASTValue lhs = children.get(0).eval(environment).tryCoerceTo(AST.ValueType.NUM);
+		ASTValue rhs = children.get(1).eval(environment).tryCoerceTo(AST.ValueType.NUM);
+
+		if((lhs.type != AST.ValueType.NUM) || (rhs.type != AST.ValueType.NUM)) {
+			throw new AssertionError("Operation arguments must both be ValueType NUM");
```

Comment:
We should only use Assertions for conditions that we expect to always be true because our implementation or the semantics of the language guarantee it.  Assertions are a way for us to ensure we haven't made an error.  In this case, however, a user could (erroneously) try to divide things that aren't numbers.  In this case, it's not an AssertionError exception, it's a WordsRuntimeException that we will eventually catch and display to the user.  You can pick from the among the existing exceptions (OperatorTypeMismatch might be a good one) or create a new one.  This comment applies not just to this method but to all of the eval arithmetic methods.